src/util/pools.ts

```diff
@@ -0,0 +1,22 @@
+import { Currency, Token } from '@uniswap/sdk-core';
+import { Pair } from '@uniswap/v2-sdk';
+
+export function getTokenWithFotTaxFromPools(token: Token, pools?: Pair[]): Token | undefined {
+  const firstPoolContainingToken = pools?.find((pool) => pool.involvesToken(token));
+  let tokenWithFotTax: Token | undefined = undefined;
+
+  if (firstPoolContainingToken) {
+    if (firstPoolContainingToken.token0.equals(token)) {
```

Comment:
mini comment heerw old help!

Reply:
Offline discussed, we had to remove the special pools filtering for reserve token with FOT tax, and instead let the quote provider calculate with fot tax populated.